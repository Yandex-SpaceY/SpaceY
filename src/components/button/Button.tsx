import React, { FC, ReactElement, MouseEvent } from 'react';
import cn from 'classnames';

import './button.scss';

interface IButton {
  onClick(e: MouseEvent<HTMLButtonElement>): void;
  className?: string;
  type?: 'button' | 'submit' | 'reset';
}

const Button: FC<IButton> = ({ onClick, className, children, type = 'button' }): ReactElement => {

  return (
    <button type={type} className={cn('one-button', className)} onClick={onClick}>{children}</button>
  );
};

export default Button;
