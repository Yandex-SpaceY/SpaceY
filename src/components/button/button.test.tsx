import * as React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

describe('test Button', () => {
  it('check render component with children', () => {
    const component = shallow(<Button>some text</Button>);

    expect(component).toMatchSnapshot();
  });
  it('check render component w/o children', () => {
    const component = shallow(<Button></Button>);

    expect(component).toMatchSnapshot();
  });
  it('check render component with type', () => {
    const component = shallow(<Button type='submit'></Button>);

    expect(component).toMatchSnapshot();
  });
  it('check render component with onClick', () => {
    const mockCallBack = jest.fn();
    const component = shallow(<Button onClick={mockCallBack}></Button>);

    expect(component).toMatchSnapshot();
  });
});
