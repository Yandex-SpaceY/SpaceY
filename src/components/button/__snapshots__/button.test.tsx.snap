// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button check render component w/o children 1`] = `
<button
  className="one-button"
  disabled={false}
  type="button"
/>
`;

exports[`test Button check render component with children 1`] = `
<button
  className="one-button"
  disabled={false}
  type="button"
>
  some text
</button>
`;

exports[`test Button check render component with onClick 1`] = `
<button
  className="one-button"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
/>
`;

exports[`test Button check render component with type 1`] = `
<button
  className="one-button"
  disabled={false}
  type="submit"
/>
`;
